cns/restserver/internalapi_linux.go

```diff
@@ -91,5 +91,40 @@ func (service *HTTPRestService) programSNATRules(req *cns.CreateNetworkContainer
 			return types.FailedToRunIPTableCmd, "[Azure CNS] failed to insert SNAT IMDS rule : " + err.Error()
 		}
 	}
+
+	// use any secondary ip + the nnc prefix length to get an iptables rule to allow dns traffic
+	// this should be idempotent if req.IPConfiguration.IPSubnet.IPAddress is an ip in the nc's subnet
+	for _, v := range req.SecondaryIPConfigs {
```

Comment:
Will this override podsubnet clusters as well. Would recommend testing these changes with a podsubnet and Overlay cluster as well such that we do not break existing SNAT rules for those.

Reply:
tested on cilium, also testing in pipelines. I believe this should be okay since the primary ip is always an ip in the pod subnet space. For the iptable rule, whether we use the primary ip or any ip in the subnet space, we should get the same network. Like 10.10.10.1/24 is the same as 10.10.10.2/24 to iptables.